### PR TITLE
[chore] Update `setup.sh`, with removed replacement script & added project-level ENVs

### PR DESCRIPTION
**Pull request by henit-chobisa**
## Tasks
- [x] Updated `setup.sh` file to split off frontend, backend and space environment variables.
- [x] Removed Replacement Script in `Start.sh`, `Dockerfile.space` and `Dockerfile.web`
- [x] Removed Environment Variables from `docker-compose.yml`, file based env variables are given
- [x] Added `.env.example` for apiserver as well

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -1,29 +1,69 @@
...
 
 echo -e "\e[1;38m 1. Head over to TipTap cloud's Pro Extensions Page, https://collab.tiptap.dev/pro-extensions \e[0m"
 echo -e "\e[1;38m 2. Copy the token given to you under the first paragraph, after 'Here it is' \e[0m \n"
 
-read -p $'\e[1;32m Please Enter Your TipTap Pro Extensions Authentication Token: \e[0m \e[1;36m' authToken
-
+read -p $'\e[1;32m Please Enter Your TipTap Pro Extensions Authentication Token: \e[0m \e[1;36m'  authToken
 
 echo "@tiptap-pro:registry=https://registry.tiptap.dev/
 //registry.tiptap.dev/:_authToken=${authToken}" > .npmrc
```
Comment (pablohashescobar) [inline setup.sh:63]: @henit-chobisa the npmrc should be also created in respective folders

File: setup.sh [hunk 2]
```diff
@@ -1,29 +1,69 @@
...
 
+
```
Comment (pablohashescobar) [inline setup.sh:65]: Also @henit-chobisa Please remove the extra spaces from bottom

File: setup.sh [hunk 3]
```diff
@@ -1,29 +1,69 @@
...
 cp ./.env.example ./.env
+echo -e "\nNEXT_PUBLIC_API_BASE_URL=$1"  >> ./.env
+echo -e "WEB_URL=$1" >> ./.env
 
 # Export for tr error in mac
 export LC_ALL=C
 export LC_CTYPE=C
 
+# Generate the SECRET_KEY that will be used by django
+echo -e "SECRET_KEY=\"$(tr -dc 'a-z0-9' < /dev/urandom | head -c50)\""  >> ./.env
```
Comment (pablohashescobar) [inline setup.sh:11]: This also can be moved to the env at `/apiserver/.env`

File: setup.sh [hunk 4]
```diff
@@ -1,29 +1,69 @@
 #!/bin/bash
 cp ./.env.example ./.env
+echo -e "\nNEXT_PUBLIC_API_BASE_URL=$1"  >> ./.env
```
Comment (pablohashescobar) [inline setup.sh:3]: Can be moved to the web and space directories let's completely remove the root `.env.example`

File: docker-compose.yml [hunk 5]
```diff
@@ -40,23 +8,8 @@ services:
       dockerfile: ./web/Dockerfile.web
       args:
         DOCKER_BUILDKIT: 1
-        NEXT_PUBLIC_API_BASE_URL: http://localhost:8000
-        NEXT_PUBLIC_DEPLOY_URL: http://localhost/spaces
     restart: always
     command: /usr/local/bin/start.sh web/server.js web
-    env_file:
```
Comment (pablohashescobar) [inline docker-compose.yml:47]: Are we not passing any environment variables here @henit-chobisa

**Review by pablohashescobar — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (henit-chobisa) [reply docker-compose.yml:47]: Not Nikhil, you see here we are directly passing the .env file from the apiserver, we don't need to pass any environment variables.

**Comment by henit-chobisa** [comment]
The above PR is coupled with the following PR #2135

**Review by sriramveeraghanta — APPROVED** [review]

**Pull request merged**